### ISSUE TITLE
[BUG] Column level names lost when constructing `DataFrame`

### ISSUE THREAD

**Issue by shwina** [rapidsai/cudf#13740]
When passing a `MultIiIndex` with level names as the `columns=` argument to a `DataFrame`, the level names are lost:

```python
import numpy as np
import cudf

columns = cudf.MultiIndex.from_tuples(
    [
        ("A", "cat"),
        ("B", "dog"),
        ("B", "cat"),
        ("A", "dog"),
    ],
    names=["exp", "animal"],
)
df = cudf.DataFrame(np.random.randn(8, 4), columns=columns)
print(columns.names)
print(df.columns.names)
```

Output:

```
['exp', 'animal']
[None]
```

**Comment by wence-** [comment]
Possibly like so:
```diff
diff --git a/python/cudf/cudf/core/dataframe.py b/python/cudf/cudf/core/dataframe.py
index 0fe8949090..318f66b62c 100644
--- a/python/cudf/cudf/core/dataframe.py
+++ b/python/cudf/cudf/core/dataframe.py
@@ -722,6 +722,10 @@ class DataFrame(IndexedFrame, Serializable, GetAttrGetItemMixin):
 
         if dtype:
             self._data = self.astype(dtype)._data
+        # Fix up
+        if isinstance(columns, pd.MultiIndex):
+            self._data.multiindex = True
+            self._data._level_names = tuple(columns.names)
 
     @_cudf_nvtx_annotate
     def _init_from_series_list(self, data, columns, index):
```

This is inappropriate intimacy, so is not a good fix. The issue with the `DataFrame` constructor is that it has ad-hoc handling of the columns argument that depends very much on the type of the `data` argument. Some code-paths look at the columns properly and produce the appropriate `ColumnAccessor`, others don't.

**Comment by galipremsagar** [comment]
Something to note, this is an issue with regular Index in various code-flows. I have a PR coming up soon to cover them

**Comment by galipremsagar** [comment]
Fixed by #13772
